app/cdap/components/DataPrep/TopPanel/index.js

```diff
@@ -382,14 +366,14 @@ export default class DataPrepTopPanel extends Component {
             <span className="text-danger">{this.state.onSubmitError}</span>
           ) : null}
           {this.state.higherVersion ? this.renderUpgradeBtn() : null}
-          {this.props.singleWorkspaceMode ? this.renderApplyBtn() : null}
-          {!this.props.singleWorkspaceMode ? (
+          {this.props.mode === 'ROUTED_WORKSPACE' ? this.renderApplyBtn() : null}
```

Comment:
For all uses of these constants: Should this be imported from the enum?

Reply:
@njbriggs I added a note in the PR description for this. Ideally we should. But for us to be able to use enum we need to change this to a tsx file. If I do that there are 100+ errors which is quite a bit of change for this PR. 

This should be done but outside of this PR.